app/code/Magento/Customer/view/frontend/web/js/show-password.js

```diff
@@ -0,0 +1,59 @@
+/**
+* Copyright © Magento, Inc. All rights reserved.
+* See COPYING.txt for license details.
+*/
+
+define([
+    'jquery',
+    'ko',
+    'uiComponent'
+], function ($, ko, Component) {
+    'use strict';
+
+    return Component.extend({
+        passwordSelector: '',
+        showPasswordSelector: '[data-role=show-password]',
+        passwordInputType: 'password',
+        textInputType: 'text',
+
+        defaults: {
+            template: 'Magento_Customer/show-password'
+        },
+
+        /** @inheritdoc */
+        initialize: function () {
+            this._super();
+        },
+
+        /**
+         * @return {Object}
+         */
+        initObservable: function () {
+            var self = this;
+
+            this._super()
+                .observe({
+                    isChecked: ko.observable(false)
+                });
+
+            this.isChecked.subscribe(function () {
+                self._showPassword();
+            });
```

Comment:
It's better to use `.bind(this)` instead of `self` variable
```suggestion
            this.isChecked.subscribe(function () {
                this._showPassword();
            }.bind(this));
```

Reply:
code is updated. thanks for letting me know this.